pkg/cmd/roachtest/tests/sysbench.go

```diff
@@ -596,6 +623,101 @@ func exportSysbenchResults(
+		t.L(),
+		func(nodeID int) (*profile.Profile, error) {
+			return roachtestutil.GetMutexProfile(ctx, c, t.L(), nodeID)
+		})
+	require.NoError(t, err)
+
+	// Similar to mutex profiles, collect the first allocs profiles.
+	allocsProfiles1, err := roachtestutil.GetProfileForAllCRDBNodes(c,
+		t.L(),
+		func(nodeID int) (*profile.Profile, error) {
+			return roachtestutil.GetAllocsProfile(ctx, c, t.L(), nodeID)
+		})
+	require.NoError(t, err)
+
+	// Collect 30 seconds of CPU profiles.
+	cpuProfiles, err := roachtestutil.GetProfileForAllCRDBNodes(c,
+		t.L(),
+		func(nodeID int) (*profile.Profile, error) {
+			return roachtestutil.GetCPUProfile(ctx, c, t.L(), nodeID, time.Second*30 /* duration */)
+		})
+	require.NoError(t, err)
+
+	// Collect the second mutex profiles.
+	mutexProfiles2, err := roachtestutil.GetProfileForAllCRDBNodes(c,
+		t.L(),
+		func(nodeID int) (*profile.Profile, error) {
+			return roachtestutil.GetMutexProfile(ctx, c, t.L(), nodeID)
+		})
+	require.NoError(t, err)
+
+	// Collect the second allocs profiles.
+	allocsProfiles2, err := roachtestutil.GetProfileForAllCRDBNodes(
+		c,
+		t.L(),
+		func(nodeID int) (*profile.Profile, error) {
+			return roachtestutil.GetAllocsProfile(ctx, c, t.L(), nodeID)
+		})
+	require.NoError(t, err)
+
+	// Merge the CPU profiles into one profile that contains all the nodes.
+	cpuMerged, err := profile.Merge(cpuProfiles)
+	require.NoError(t, err)
+
+	// Calculate the difference between the two mutex profiles, and also between
+	// the two allocs profiles.
+	mutexDiffProfiles := make([]*profile.Profile, len(c.CRDBNodes()))
+	allocsDiffProfiles := make([]*profile.Profile, len(c.CRDBNodes()))
+	for i := range len(c.CRDBNodes()) {
+		mutexProfiles1[i].Scale(-1)
+		mutexDiffProfiles[i], err = profile.Merge([]*profile.Profile{mutexProfiles1[i],
+			mutexProfiles2[i]})
+		require.NoError(t, err)
+		mutexDiffProfiles[i].TimeNanos = mutexProfiles2[i].TimeNanos
+		mutexDiffProfiles[i].DurationNanos = mutexProfiles2[i].TimeNanos - mutexProfiles1[i].TimeNanos
+
+		allocsProfiles1[i].Scale(-1)
+		allocsDiffProfiles[i], err = profile.Merge([]*profile.Profile{allocsProfiles1[i],
+			allocsProfiles2[i]})
+		require.NoError(t, err)
+		allocsDiffProfiles[i].TimeNanos = allocsProfiles2[i].TimeNanos
+		allocsDiffProfiles[i].DurationNanos = allocsProfiles2[i].TimeNanos -
+			allocsProfiles1[i].TimeNanos
+	}
+
+	// Merge the mutex diff profiles into one profile that contains all the nodes.
+	mutexDiffMerged, err := profile.Merge(mutexDiffProfiles)
+	require.NoError(t, err)
+
+	// Do the same for the allocs diff profiles.
+	allocsDiffMerged, err := profile.Merge(allocsDiffProfiles)
+	require.NoError(t, err)
+
+	// exportProfile exports a profile to the artifacts' directory.
+	exportProfile := func(profile *profile.Profile, filename string) {
+		buf := bytes.Buffer{}
+		require.NoError(t, profile.Write(&buf))
+		err = os.WriteFile(filepath.Join(t.ArtifactsDir(), filename), buf.Bytes(), 0644)
```

Comment:
Let's put these into `filepath.Join(t.PerfArtifactsDir(), "profiles")`. If they got uploaded to our nightly bucket and preserved[^1], this may come in handy - we can go back and track the change in profiles over time if we ever want to do that.
I think all we need for that is this little one-line change in in roachtest_util.sh:

```
      done <<< "$(find . -type f \( -name "${stats_file_name}" -o -name '*.pb.gz' \) | sed 's/^\.\///')")
```

[^1]: https://github.com/cockroachdb/cockroach/blob/7262f5ce8ea9080afc173a61c2b8a614937a3b6f/build/teamcity/util/roachtest_util.sh#L75-L95